### PR TITLE
Refactored how we establish the page's referrer acquisition data

### PR DESCRIPTION
**Pull request by paulbrown1982**
Refactored how we establish the page's referrer acquisition data. Mostly renames and functionisation. 

The main difference in logic is a fix where presence of a `&gclid=` value now clears all other referrer acquisition data and sets its own campaign code too.

- [x] Tested locally

### CODE REVIEW

File: support-frontend/assets/helpers/tracking/acquisitions.js [hunk 1]
```diff
@@ -283,16 +274,54 @@ function deserialiseReferralData(serialised: string): Object {
   };
 }
 
+
+// Reads the acquisition data from sessionStorage.
+function getReferrerAcquisitionDataFromSessionStorage(): ?ReferrerAcquisitionData {
```
Comment (paulbrown1982) [inline support-frontend/assets/helpers/tracking/acquisitions.js:279]: This is a copy-paste and rename of the deleted `readReferrerAcquisitionData` above.

**Review by tomrf1 — APPROVED** [review]
👍 looks good.
I'm starting to think we need unit tests for this file...

**Comment by paulbrown1982** [comment]
I've created a card for further improvements to `acquisitions.js`, including refactoring so that this stuff is more unit testable.

File: support-frontend/assets/helpers/tracking/acquisitions.js [hunk 2]
```diff
@@ -283,16 +274,54 @@ function deserialiseReferralData(serialised: string): Object {
...
+  return null;
+}
+
+// Generates appropriate acquisition data from the various, known, PPC params
+function getAcquisitionDataFromPPCParams(): ?Object {
+  if (getQueryParameter('gclid')) {
+    return {
+      source: 'PPC',
+      campaignCode: 'Google_Adwords',
+      queryParameters: getAllQueryParams(),
```
Comment (johnduffell) [inline support-frontend/assets/helpers/tracking/acquisitions.js:306]: this line caused a backend serialisation error for users with this kind of ID and has been fixed in this PR https://github.com/guardian/support-frontend/pull/3222

**Pull request merged**